src/mixins/allowable-actions-mixin.ts

```diff
@@ -2,101 +2,158 @@ import Vue from 'vue'
+        const isNotaDcRoute = !(this.$route.matched.some(route => route.name === Routes.DIGITAL_CREDENTIALS))
 
-        return isDissolveAllowed
+        return (ff && isNotaDcRoute && this.isGoodStanding && this.isBComp && !this.isRoleStaff)
       }
 
-      case AllowableActions.DOWNLOAD_BUSINESS_SUMMARY: {
-        return (!!businessId &&
-          !!getFeatureFlag('supported-business-summary-entities')?.includes(this.getLegalType))
+      case AllowableActions.DIRECTOR_CHANGE: {
+        if (this.isFirm) {
+          return this.isAllowedFiling(FilingTypes.CHANGE_OF_REGISTRATION)
+        }
+        return this.isAllowedFiling(FilingTypes.CHANGE_OF_DIRECTORS)
       }
 
-      case AllowableActions.EDIT_BUSINESS_PROFILE: {
-        return !!businessId // FUTURE: disable for staff? see #14314
+      case AllowableActions.FREEZE_UNFREEZE: {
+        return this.isAllowedFiling(FilingTypes.ADMIN_FREEZE)
       }
 
-      case AllowableActions.FILE_ADDRESS_CHANGE: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId)
+      case AllowableActions.LIMITED_RESTO_CONVERT: {
+        return this.isAllowedFiling(FilingTypes.RESTORATION, FilingSubTypes.RESTORATION_CONVERSION)
       }
 
-      case AllowableActions.FILE_ANNUAL_REPORT: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId)
+      case AllowableActions.LIMITED_RESTO_EXTEND: {
+        return this.isAllowedFiling(FilingTypes.RESTORATION, FilingSubTypes.RESTORATION_EXTENSION)
       }
 
-      case AllowableActions.FILE_CORRECTION: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId && this.isRoleStaff)
+      case AllowableActions.PUT_BACK_ON: {
+        return this.isAllowedFiling(FilingTypes.PUT_BACK_ON)
       }
 
-      case AllowableActions.FILE_DIRECTOR_CHANGE: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId)
+      case AllowableActions.RECORD_CONVERSION: {
+        return this.isAllowedFiling(FilingTypes.CONVERSION)
       }
 
-      case AllowableActions.FILE_STAFF_NOTATION: {
-        return (!this.isHistorical && !this.hasBlockerExceptStaffApproval && !!businessId && this.isRoleStaff)
+      case AllowableActions.REGISTRARS_NOTATION: {
+        return this.isAllowedFiling(FilingTypes.REGISTRARS_NOTATION)
       }
 
-      case AllowableActions.VIEW_CHANGE_COMPANY_INFO: {
-        const isAllowedEntityType = (
-          this.isBenBcCccUlc ||
-          (this.isCoop && !!getFeatureFlag('special-resolution-ui-enabled')) ||
-          this.isPartnership ||
-          this.isSoleProp
+      case AllowableActions.REGISTRARS_ORDER: {
+        return this.isAllowedFiling(FilingTypes.REGISTRARS_ORDER)
+      }
+
+      case AllowableActions.RESTORATION: {
+        return (
+          this.isAllowedFiling(FilingTypes.RESTORATION, FilingSubTypes.RESTORATION_FULL) ||
+          this.isAllowedFiling(FilingTypes.RESTORATION, FilingSubTypes.RESTORATION_LIMITED)
         )
```

Comment:
There is still code that uses this case. Ideally, in the future, code would look at only FullRestoration and LimitedRestoration cases.

This is actually even incorrect, since there are 2 other restoration cases that aren't included here. Oh well.